NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h

```diff
@@ -106,6 +106,75 @@ inline void vectorFill( float* result, float value, int vectorSize )
 	}
 }
 
+inline void vectorFill0( float* result, int vectorSize )
+{
+	int sseSize;
+	int nonSseSize;
+	checkSse(vectorSize, sseSize, nonSseSize);
+
+	__m128 valueSse = _mm_setzero_ps();
+
+	while( sseSize >= 4 ) {
+		_mm_storeu_ps(result, valueSse);
+		result += 4;
+		_mm_storeu_ps(result, valueSse);
+		result += 4;
+		_mm_storeu_ps(result, valueSse);
+		result += 4;
+		_mm_storeu_ps(result, valueSse);
+		result += 4;
+
+		sseSize -= 4;
+	}
+
+	while( sseSize > 0 ) {
+		_mm_storeu_ps(result, valueSse);
+		result += 4;
+		sseSize--;
+	}
+}
```

Comment:
Is `vectorFill0` used only when `vectorSize % 4 == 0`?

If it's not then you must fill last `nonSseSize` elements with zeroes as well

Reply:
you are right, it should be used for any vectorSize